### DISCUSSION TITLE
Laravel/Inertia/Vue3 Ray usage using composition api

### DISCUSSION THREAD

**Discussion by computer-logix** [spatie/ray#684] [Q&A]
With the app.js below Ray works fine on the PHP side but doesn't work on the Vue side even with permafrost-dev/vue-ray installed.
Does anyone have a code fragment to demonstrate a working configuration?

**app.js**
```
import {createApp, h} from 'vue';
import {createInertiaApp} from '@inertiajs/inertia-vue3';
import {InertiaProgress} from '@inertiajs/progress';
import PortalVue from 'portal-vue';
import VueClipboard from 'vue3-clipboard';
import {store} from './store';

const { RayPlugin } = require('vue-ray');
const appName = window.document.getElementsByTagName('title')[0]?.innerText || 'Laravel';

createInertiaApp({
  title: (title) => `${title} - ${appName}`,
  resolve: (name) => require(`./Pages/${name}.vue`),
  setup({el, app, props, plugin}) {
    return createApp({render: () => h(app, props)})
      .use(PortalVue)
      .use(VueClipboard)
      .use(plugin)
      .use(store)
      .use(RayPlugin, {
        interceptErrors: true,
        host: '127.0.0.1',
        port: 23517,
        nodeRaySettings: {
          interceptConsoleLog: false,
        },
      })
      .mixin({methods: {route}})
      .mount(el);
  },
});

InertiaProgress.init({color: '#4B5563'});
```

**Comment by patinthehat** [comment]
@computer-logix Can you provide an example of where it's not working?

**Comment by iamlasse** [comment]
You can't use "this" in vue3 setup scripts, so what should you use instead to access $ray method?

**Reply by patinthehat** [reply]
I am working on enabling the [vue-ray](https://github.com/permafrost-dev/vue-ray) package to work with the composition api in vue 3.  Stay tuned.

In the mean time, you may use [node-ray](https://github.com/permafrost-dev/node-ray) for interacting with Ray in JS/TS.

**Reply by lefrancois** [reply]
Hey @patinthehat, any idea when the composition api is supported?

**Reply by patinthehat** [reply]
@lefrancois Currently the feature is on hold as I'm busy with work, but the goal is to implement it in the coming months.

**Reply by lefrancois** [reply]
@patinthehat Thanks for the info! We started using the node version now, but would be happy to have a vue version. Have a good time and thanks for your work!

**Comment by patinthehat** [comment]
vue-ray 2.0 with composition api support is being prepped for release in the next few weeks and should resolve your issue/use case. Stay tuned!

**Comment by patinthehat** [comment]
@computer-logix @lefrancois `vue-ray` 2.0 with composition api support [was released](https://github.com/permafrost-dev/vue-ray/releases/tag/v2.0.1)!
